news-bundle/src/Event/FetchArticlesForFeedEvent.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\Event;
+
+use Contao\Model\Collection;
+use Contao\PageModel;
+use FeedIo\FeedInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class FetchArticlesForFeedEvent
+{
+    public const NAME = 'contao_news.feed.fetch_articles';
+
+    private ?Collection $articles = null;
+
+    public function __construct(private readonly FeedInterface $feed, private readonly Request $request, private readonly PageModel $pageModel)
+    {
+    }
+
+    public function getFeed(): FeedInterface
+    {
+        return $this->feed;
+    }
+
+    public function getRequest(): Request
+    {
+        return $this->request;
+    }
+
+    public function getPageModel(): PageModel
+    {
+        return $this->pageModel;
+    }
+
+    public function getArticles(): ?Collection
```

Comment:
I think we should use an array of models here instead of a `Collection`, that would be more flexible to add own articles. And maybe a `addArticle` method as well?

Reply:
Fixed in 9455e015c